### DISCUSSION TITLE
Copy-paste channel

### DISCUSSION THREAD

**Discussion by stkxray** [ImageMagick/ImageMagick#3821] [Help]
Hi!
How to copy the alpha channel from one image and paste into the red channel of another image? 
Thanks in advance!

**Comment by fmw42** [comment]
IM6 Unix syntax. The following will copy the alpha channel from image2 and put it into the red channel of image1.

`convert image1 \( image2 -alpha extract \) -compose copy_red -composite new_image1`

If on Windows, remove the \ from the parentheses. If on IM 7, change convert to magick.

Please always provide your IM version and platform in the future.

**Comment by urban-warrior** [comment]
For ImageMagick 7, try the `-channel-fx` option @ https://imagemagick.org/script/command-line-options.php#channel-fx.

**Reply by stkxray** [reply]
**IM version/platform:** 7/Win10x64
I don’t know what I’m doing wrong, but it doesn’t work:
```
magick bump.dds -channel-fx alpha=>red nmap.tga 
magick bump.dds -channel-fx blue=>green nmap.tga 
magick bump.dds -channel-fx green=>blue nmap.tga 
```
